### PR TITLE
External target for netem and partition chaos

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: Yang Keao <keao.yang@yahoo.com>

### What problem does this PR solve?

#676 , you can see an example here:

```
apiVersion: pingcap.com/v1alpha1
kind: NetworkChaos
metadata:
  name: network-delay-example
  namespace: chaos-testing
spec:
  action: delay
  mode: one
  selector:
    labelSelectors:
      "app.kubernetes.io/component": "tikv"
  delay:
    latency: "90ms"
    correlation: "25"
    jitter: "90ms"
  direction: to
  externalTargets:
    - "8.8.8.8"
    - "www.google.com"
    - "8.8.0.0/16"
  duration: "10s"
  scheduler:
    cron: "@every 15s"
```

```
apiVersion: pingcap.com/v1alpha1
kind: NetworkChaos
metadata:
  name: network-partition-example
  namespace: chaos-testing
spec:
  action: partition
  mode: one
  selector:
    labelSelectors:
      "app.kubernetes.io/component": "tikv"
  direction: to
  externalTargets:
    - "8.8.8.8"
    - "8.8.0.0/16"
    - "www.google.com"
  duration: "10s"
  scheduler:
    cron: "@every 15s"
```

### What is changed and how does it work?

Add an `externalTarget` field to specify external endpoints as targets. This extra field was used to avoid breaking compatibility, in the future we may need to redesign these fields.

### Check List

Tests

I have tested this feature manually. With a simple ubuntu pod and `ping` external targets, everything works well.

Code changes

- Has Go code change

Related changes

- Need to update the documentation

```release-note
Support external target for netem and partition chaos
```

### CODE REVIEW

File: api/v1alpha1/networkchaos_types.go [hunk 1]
```diff
@@ -166,6 +166,10 @@ type NetworkChaosSpec struct {
 	// Target represents network target, this applies on netem and network partition action
 	// +optional
 	Target *Target `json:"target,omitempty"`
+
+	// ExternalTargets represents network targets outside k8s
+	// +optional
+	ExternalTargets []string `json:"externalTargets,omitempty"`
```
Comment (cwen0) [inline api/v1alpha1/networkchaos_types.go:172]: Can we use some way to represent the range of IPs? such as "192.168.0.0/24"

Comment (YangKeao) [reply api/v1alpha1/networkchaos_types.go:172]: It supports CIDR input now :smile:

File: controllers/networkchaos/netem/utils.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 PingCAP, Inc.
```
Comment (fewdan) [inline controllers/networkchaos/netem/utils.go:1]: s/2019/2020/g

File: controllers/networkchaos/netutils/cidr.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 PingCAP, Inc.
```
Comment (fewdan) [inline controllers/networkchaos/netutils/cidr.go:1]: ditto

File: helm/chaos-mesh/values.yaml [hunk 4]
```diff
@@ -21,7 +21,7 @@ controllerManager:
   replicaCount: 1
 
   image: pingcap/chaos-mesh:latest
-  imagePullPolicy: Always
+  imagePullPolicy: Never
```
Comment (cwen0) [inline helm/chaos-mesh/values.yaml:24]: Fix this?

Comment (YangKeao) [reply helm/chaos-mesh/values.yaml]: Fixed

**Pull request merged**